flink-connector-debezium/src/main/java/com/alibaba/ververica/cdc/debezium/table/RowDataDebeziumDeserializeSchema.java

```diff
@@ -110,11 +110,17 @@ public void deserialize(SourceRecord record, Collector<RowData> out) throws Exce
 		} else {
 			GenericRowData before = extractBeforeRow(value, valueSchema);
 			validator.validate(before, RowKind.UPDATE_BEFORE);
-			before.setRowKind(RowKind.UPDATE_BEFORE);
-			out.collect(before);
 
 			GenericRowData after = extractAfterRow(value, valueSchema);
 			validator.validate(after, RowKind.UPDATE_AFTER);
+			// ignore the record on no change fields
+			if (after.equals(before)) {
```

Comment:
I just find that the equal could be heavy if there are lots of columns, because this is a per-record operation. What about we introduce back the option and disable the compare operation by default? 

I would suggest `capture-unchanged-updates` option which is `true` by default and can enabel manually. What do you think?

Reply:
I thought about this question today, I am still going to discuss it with you, and I think it should be like this

Reply:
if we can get the updated column names, it will be easy, I will try to find it